staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -212,6 +217,27 @@ func newAuthenticator(c *cache, config *api.ExecConfig, cluster *clientauthentic
 	return c.put(key, a), nil
 }
 
+func isInteractive(isTerminalFunc func(int) bool, config *api.ExecConfig) (bool, error) {
+	if config.InteractiveMode == nil {
+		// backwards compatible default
+		return isTerminalFunc(int(os.Stdout.Fd())), nil
+	}
+
+	switch *config.InteractiveMode {
+	case api.NeverExecInteractiveMode:
+		return false, nil
```

Comment:
I checked and I don't think we can do set `stdin` to an `io.Reader` that detects if a `Read()` took place in order to provide the user with a nice error message. This is because the go standard library tried to read that `io.Reader` in a loop and so it will _always_ get `Read()`, even if the exec plugin on the other end of the pipe does not `Read()`.

https://github.com/golang/go/blob/0f66fb7b856b02497cf801ce72d80f375f53358b/src/os/exec/exec.go#L266

Reply:
meh, that is a bummer if true.

TODO: look into this more next week.

Reply:
I do see some special handling of `*os.File`, maybe we could make one using `os.NewFile` and try to see if that got read?

Reply:
I made a helper program to try this out. The helper program is really messy, but it should run and teach us something. https://gist.github.com/ankeesler/7173d29d4ac81c8b97b6c584836db142

Here is what I learned:
- If a parent process opens a pipe and passes it to a child process, it can read from that pipe to determine whether the child process has read from the pipe (I think I convinced myself of this since the child process gets an exact copy of the parent's file descriptor table, so that same pipe file descriptor will have the same seek offset for both processes)

So my next question is...if the parent process writes something to a pipe and then passes the read end to the child process, then how can the child process tell the difference between something typed from the user and a sentinel value from the parent process?

For example, if the parent process was to write a single byte to the write end of the pipe, and then the child process read from the pipe, then the child process would get a single byte, which they could suspect came from the user.

Reply:
Correct me if I am wrong, but on Zoom we discussed punting on this for now.